R/extract_diagnostics.R

```diff
@@ -0,0 +1,161 @@
+#' Extract diagnostic metrics from model fit and data
+#'
+#' This function extracts various diagnostic metrics from a fitted `EpiNow2`
+#' model and provided data. It checks for low case counts and computes
+#' diagnostics from the fitted model, including the mean acceptance
+#' statistic, divergent transitions, maximum tree depth, and Rhat values.
+#' These diagnostics are then flagged if they exceed specific thresholds,
+#' and the results are returned as a data frame.
+#'
+#' @param fit A list containing the model fit object from `EpiNow2`, which
+#'            includes `estimates$fit`.
+#' @param data A data frame containing the input data used in the model fit.
```

Comment:
+1 for it being better to extract objects from the `fit` all else equal

Reply:
The provided dataset is only used by the `low_case_count_diagnostic()`. I think one could make a reasonable argument either way on parsing that particular metric based on the EpiNow2 returned dataset and our input dataset.

(ignore this part if you don't want to read my musings)

Thinking it through: let's say there's some kind of parsing issue between the specified data and EpiNow2 and we rely on the EpiNow2 dataset. Then, we include/exclude a model fit that we would have excluded/included based on the input data (function as written). Is that correct?

In some sense, it is correct because the metric is correctly reflecting the data read by the model. In another sense, it's wrong because it doesn't reflect the case data passed to the model.

I don't think one or the other would be an easier bug to catch because the input data format passed to EpiNow2 is explicitly tested. Maybe if we were running multiple models we would notice when one was generated and another wasn't? But I don't find that particularly convincing.

-----------------

Ok I don't feel super strongly but I don't think it's more foolproof to rely on EpiNow2's internal format. I think it comes down to whether we think the model's internal state is more "correct" for external flags than our knowledge of reality. Aaaaand re-reading that I may be too deep in this rabbit hole.

Reply:
Basically I don't feel super strongly but I don't love relying on EpiNow2 for something that isn't EpiNow2 specific. This metric is only for external reporting, so we could reasonably not include it here and only keep it with postprocessing.